Realm/RLMRealm.mm

```diff
@@ -183,34 +183,93 @@ + (instancetype)realmWithURL:(NSURL *)fileURL {
     return [RLMRealm realmWithConfiguration:configuration error:nil];
 }
 
+// The server doesn't send us the subscriptions for permission types until the
+// first subscription is created. This is fine for synchronous opens (if we're
+// creating a new Realm we create the permission objects ourselves), but it
+// causes issues for asyncOpen because it means that when our download completes
+// we don't actually have the full Realm state yet.
+static void waitForPartialSyncSubscriptions(Realm::Config const& config) {
+    auto realm = Realm::get_shared_realm(config);
+    auto table = ObjectStore::table_for_object_type(realm->read_group(), "__ResultSets");
```

Comment:
If they are available here, prefer the constants from ObjectStore: `realm::partial_sync::result_sets_type_name` and below `realm::partial_sync::property_expires_at`